consensus/src/pipeline/buffer_manager.rs

```diff
@@ -333,6 +343,48 @@ impl BufferManager {
         }
     }
 
+    fn try_add_pending_commit_vote(&mut self, vote: CommitVote) -> bool {
+        let block_id = vote.commit_info().id();
+        let round = vote.commit_info().round();
+        let epoch = vote.commit_info().epoch();
+
+        // Store the commit vote only if it is for one of the next 100 rounds in the same epoch.
+        // As the buffer manager is reset between epoch, we don't benefit from storing the pending commit votes
+        // for the next epoch.
+        if epoch == self.epoch_state.epoch
+            && round > self.highest_committed_round
+            && self.highest_committed_round + self.max_pending_rounds_in_commit_vote_cache > round
+        {
+            if let Some(votes) = self.pending_commit_votes.get_mut(&round) {
+                if self
+                    .epoch_state
+                    .verifier
+                    .get_voting_power(&vote.author())
+                    .is_some()
```

Comment:
I don't think we need this given that the signature was verified (it can't verify without being in the validator set)?

Reply:
The commit vote verification checks that the address is in validator set, but doesn't check it has non zero voting power. But checking for non zero voting power just seems a minor check that can be removed.
With optimistic signature verification, the votes may not be verified. So, we will have to put check again to avoid OOM attacks.

Reply:
what does optimistic verification have to do with this? we only allow one vote per author per round regardless of signature?

Reply:
But what if a malicious party submits many votes under different author names who either don't exist in ValidatorVerifier or has a zero voting power? That will be an OOM attack vector too?

Reply:
good point, we should have the check that the sender matches the declared author field, the network link is encrypted with noise handshake and the key is on-chain. so after we move to optimistic verification, we can add the sender check here https://github.com/aptos-labs/aptos-core/blob/6a71f18b8091e67ff567fe47715176c06a508917/consensus/src/network.rs#L720